app/services/room_settings_getter.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class RoomSettingsGetter
+  # Special options are meeting options that have different values other than `true|false` to represent postives and negatives.
+  # The `special_options` hash is a registry that keeps hold of all of the special options used in GL3 along with their postive and negative values.
+  # When adding a new special option just register it in this Hash respecting the fallowing format:
+  # Hash(`<option_name> => {'true' => <Postive>, 'false' => <Negative>})`
+  SPECIAL_OPTIONS = { 'guestPolicy' => { 'true' => 'ASK_MODERATOR', 'false' => 'ALWAYS_ACCEPT' } }.freeze
+
+  def initialize(room_id:, provider:, only_bbb_options: false)
+    @room_id = room_id
+    @only_bbb_options = only_bbb_options
+    @rooms_configs = MeetingOption.joins(:rooms_configurations)
+                                  .where(rooms_configurations: { provider: })
+                                  .pluck(:name, :value)
+                                  .to_h
+  end
+
+  def call
+    room_settings = MeetingOption.joins(:room_meeting_options).where(room_meeting_options: { room_id: @room_id })
```

Comment:
This is a tiny optimization that simplifies the logic by removing the need to duplicate two complete queries per case or to use multiple variables for each query relation.